### PR TITLE
Fix qDigest values_at_quantiles function to throw INVALID_FUNCTION_ARGUMENT on null input

### PR DESCRIPTION
**Pull request by duxiao1212**
When a `null` value exists in the qDigest `values_at_quantiles` function, we currently treat it as 0.0, which can lead to confusion and errors. 
Instead, we should throw an error when encountering a null value in the parameter, ensuring consistency with the behavior of value_at_quantile in both Tdigest and Qdigest, as well as values_at_quantiles in [Tdigest](http://github.com/prestodb/presto/pull/24969)

## Description
addd null check in values_at_quantiles

## Motivation and Context
bug fix 

## Impact
low impact

## Test Plan
unit test

## Contributor checklist

- [ ] Please make sure your submission complies with our [contributing guide](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md), in particular [code style](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md#code-style) and [commit standards](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md#commit-standards).
- [ ] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-main-base/src/test/java/com/facebook/presto/operator/scalar/TestQuantileDigestFunctions.java [hunk 1]
```diff
@@ -60,6 +60,24 @@ public void testGetValueAtQuantileBelowZero()
...
+                format("values_at_quantiles(CAST(X'%s' AS qdigest(bigint)), ARRAY[0.25, NULL, 0.75])",
+                        toHexString(qdigest)),
+                BIGINT,
+                null);
+
+        functionAssertions.assertFunction(
+                format("values_at_quantiles(CAST(X'%s' AS qdigest(double)), ARRAY[0.25, NULL, 0.75])",
+                        toHexString(qdigest)),
+                DOUBLE,
+                null);
```
Comment (spershin) [inline presto-main-base/src/test/java/com/facebook/presto/operator/scalar/TestQuantileDigestFunctions.java]: Maybe also add a REAL case too, since we have 3 flavors?

**Review by spershin — APPROVED** [review]
Looks good.
It will still need a committer's review.

File: presto-main-base/src/test/java/com/facebook/presto/operator/scalar/TestQuantileDigestFunctions.java (hunk 1 above)
Comment (duxiao1212) [reply presto-main-base/src/test/java/com/facebook/presto/operator/scalar/TestQuantileDigestFunctions.java]: thanks. updated and refactored into 3 functions.

File: presto-main-base/src/main/java/com/facebook/presto/operator/scalar/QuantileDigestFunctions.java [hunk 2]
```diff
@@ -134,6 +136,7 @@ public static Block valuesAtQuantilesBigint(@SqlType("qdigest(bigint)") Slice in
         QuantileDigest digest = new QuantileDigest(input);
         BlockBuilder output = BIGINT.createBlockBuilder(null, percentilesArrayBlock.getPositionCount());
         for (int i = 0; i < percentilesArrayBlock.getPositionCount(); i++) {
+            checkCondition(!percentilesArrayBlock.isNull(i), INVALID_FUNCTION_ARGUMENT, "All quantiles should be non-null.");
```
Comment (kaikalur) [inline presto-main-base/src/main/java/com/facebook/presto/operator/scalar/QuantileDigestFunctions.java]: You might want do this only if mayHaveNulls returns true - small optimization. But good habit to develop :)

File: presto-main-base/src/test/java/com/facebook/presto/operator/scalar/TestQuantileDigestFunctions.java [hunk 3]
```diff
@@ -60,6 +61,37 @@ public void testGetValueAtQuantileBelowZero()
...
+    }
+
+    private void assertQuantileDigestThrowsForNulls(String typeName, Type type)
+    {
+        QuantileDigest qdigest = new QuantileDigest(1);
+        functionAssertions.assertFunction(
+                format("values_at_quantiles(CAST(X'%s' AS qdigest(%s)), ARRAY[0.25, NULL, 0.75])",
+                        toHexString(qdigest), typeName),
+                type,
+                null);
```
Comment (kaikalur) [inline presto-main-base/src/test/java/com/facebook/presto/operator/scalar/TestQuantileDigestFunctions.java:93]: Also add another test case where all elements are null.

**Review by kaikalur — CHANGES_REQUESTED** [review]

File: presto-main-base/src/main/java/com/facebook/presto/operator/scalar/QuantileDigestFunctions.java (hunk 2 above)
Comment (duxiao1212) [reply presto-main-base/src/main/java/com/facebook/presto/operator/scalar/QuantileDigestFunctions.java]: That's worth a 'Today I Learned', thb , thanks for pointing it out!

File: presto-main-base/src/main/java/com/facebook/presto/operator/scalar/QuantileDigestFunctions.java [hunk 4]
```diff
@@ -120,7 +124,11 @@ public static Block valuesAtQuantilesReal(@SqlType("qdigest(real)") Slice input,
     {
         QuantileDigest digest = new QuantileDigest(input);
         BlockBuilder output = REAL.createBlockBuilder(null, percentilesArrayBlock.getPositionCount());
+        final boolean mayHaveNull = percentilesArrayBlock.mayHaveNull();
         for (int i = 0; i < percentilesArrayBlock.getPositionCount(); i++) {
+            if (mayHaveNull) {
```
Comment (kaikalur) [inline presto-main-base/src/main/java/com/facebook/presto/operator/scalar/QuantileDigestFunctions.java]: Not quite - you want to do this check outside the loop so you will have two versions - one null checking and the other not. See some other arrays udfs for ideas

**Review by kaikalur — CHANGES_REQUESTED** [review]
See the comment

**Review by kaikalur — APPROVED** [review]

**Review by rschlussel — APPROVED** [review]
